src/observations/test_price_observation.py

```
@@ -0,0 +1,100 @@
+import pytest
+import numpy as np
+import pandas as pd
+from src.observations.price_observation import PriceObservation, PriceEnvironment
+
+
+# This environment mocks the PriceEnvironment protocol class
+@pytest.fixture
+def mock_env():
+    env = Mock(spec=TrendEnvironment)
```

Ruff

Undefined name `Mock`

(F821)